### PR TITLE
Fixes for GCC 5.x build

### LINKED ISSUE #858: Support ARM GCC 5.x series
When compiling with the 5.x series of ARM GCC, there are new compiler errors that don't appear with the 4.9.x series. 

@sergeuz  is investigating this.

### PR DESCRIPTION
**Pull request by sergeuz**
Fixes #858

### CODE REVIEW

File: build/common-tools.mk [hunk 1]
```diff
@@ -10,7 +10,7 @@ MKDIR = mkdir -p
 
 CC = $(GCC_ARM_PATH)$(GCC_PREFIX)gcc
 CPP = $(GCC_ARM_PATH)$(GCC_PREFIX)g++
-AR = $(GCC_ARM_PATH)$(GCC_PREFIX)ar
+AR = $(GCC_ARM_PATH)$(GCC_PREFIX)gcc-ar
```
Comment (sergeuz) [inline build/common-tools.mk]: Regular `ar` started to complain about missing plugins. `gcc-ar` wrapper supplies `ar` with all necessary paths.

File: hal/inc/pinmap_hal.h [hunk 2]
```diff
@@ -99,8 +99,6 @@ STM32_Pin_Info* HAL_Pin_Map(void);
 
 // todo - this is corev1 specific, needs to go in a conditional define
 
-#define LED1 LED_USER
```
Comment (sergeuz) [inline hal/inc/pinmap_hal.h]: Probably not a best fix possible, but linker started to generate warning (or error in our case) due to `LED1` define breaking ODR during LTO. `LED1` value is already defined as part of `Led_TypeDef` enum.

File: platform/NET/build.mk [hunk 3]
```diff
@@ -5,3 +5,6 @@
 
 PLATFORM_NET_PATH = $(PLATFORM_MODULE_PATH)/NET/$(PLATFORM_NET)
 include $(call rwildcard,$(PLATFORM_NET_PATH)/,sources.mk)
+
+# Stick to some POSIX-conforming API to disable BSD extensions
+CFLAGS += -D_POSIX_C_SOURCE=200809
```
Comment (sergeuz) [inline platform/NET/build.mk]: `time.h` started to include `select.h` (indirectly) which contains declarations conflicting with CC3000. Thankfully, it's guarded by `__BSD_VISIBLE` which is not defined when POSIX-compliant API is requested explicitly.

**Comment by m-mcgowan** [comment]
Great job! Can't wait to try it out see if any size-reduction advances have been made with the new compiler versions.

**Comment by m-mcgowan** [comment]
Doing a clean build for the Core I get these errors:

```
re/core_subsys_hal.c
In file included from /Users/mat1/Downloads/gcc-arm-none-eabi-5_2-2015q4/arm-none-eabi/include/sys/select.h:27:0,
                 from /Users/mat1/Downloads/gcc-arm-none-eabi-5_2-2015q4/arm-none-eabi/include/sys/types.h:89,
                 from /Users/mat1/Downloads/gcc-arm-none-eabi-5_2-2015q4/arm-none-eabi/include/stdio.h:48,
                 from src/core/core_subsys_hal.c:28:
/Users/mat1/Downloads/gcc-arm-none-eabi-5_2-2015q4/arm-none-eabi/include/sys/_timeval.h:51:8 error: redefinition of 'struct timeval'
 struct timeval {
        ^
In file included from ../platform/NET/CC3000/CC3000_Host_Driver/nvmem.h:38:0,
                 from src/core/core_subsys_hal.c:27:
../platform/NET/CC3000/CC3000_Host_Driver/cc3000_common.h:166:16: note: originally defined here
 typedef struct timeval timeval;
                ^
```

**Comment by m-mcgowan** [comment]
The error above is fixed if you move these lines

```
# Stick to some POSIX-conforming API to disable BSD extensions
CFLAGS += -D_POSIX_C_SOURCE=200809
```

from `build.mk` to `import.mk` in that project, so that any other projects importing the header files also adhere to that posix version.

**Comment by sergeuz** [comment]
oops, seems I forgot to commit `hal/src/core/tlm.mk`. Thanks for fixing that

**Pull request merged**